packages/react-router/src/routeRegistry.ts

```diff
@@ -66,22 +65,40 @@ export class RouteRegistry {
 
         // Create a new array so the routes array is immutable.
         this.#routes = [...this.#routes, ...routes];
```

Comment:
This works because node.js is single threaded, right? This codepath can't be simultaneously executed for multiple module registrations at the same time

Reply:
Yes browsers are single thread.

Furthermore there is a validation making sure that `registerLocalModules` and `registerRemoteModules` are only called once.